src/RpcServer/RelayActor.cs

```diff
@@ -0,0 +1,55 @@
+using Akka.Actor;
+using Neo.Network.P2P.Payloads;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using static Neo.Ledger.Blockchain;
+
+namespace Neo.Plugins
+{
+    public class RelayActor : UntypedActor
+    {
+        private readonly NeoSystem neoSystem;
+        private readonly Dictionary<UInt256, IActorRef> senders = new Dictionary<UInt256, IActorRef>();
+        private readonly int expireMs;
+
+        public RelayActor(NeoSystem neoSystem, int expireMs)
+        {
+            this.neoSystem = neoSystem;
+            this.expireMs = expireMs;
+            Context.System.EventStream.Subscribe(Self, typeof(RelayResult));
+        }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case IInventory inventory:
+                    {
+                        UInt256 hash = inventory.Hash;
+                        senders.Add(hash, Sender);
+                        neoSystem.Blockchain.Tell(inventory);
+                        // the sender will clean the record after expireMs
+                        Task.Run(async () =>
+                        {
```

Comment:
I prefer only one task and check the whole list every X seconds, because one task per inventory could be slower.

Reply:
Or maybe we can have a list with max X items, and trim it if we exceed this limit. I am doing this class for oracles

Reply:
please review it again.